satpy/scene.py

```diff
@@ -810,14 +806,9 @@ def resample(self, destination=None, datasets=None, generate=True,
                 arguments.
 
         """
-        to_resample_ids = [dsid for (dsid, dataset) in self._datasets.items()
-                           if (not datasets) or dsid in datasets]
-
```

Comment:
by removing this, you potentially change the default destination area. Eg, if a ABI scene has B02 and B04, and you just ask to resample B04, the destination will be the B02 area. Is this intentional?

Reply:
Why would it be the B02 area? You mean if the `finest_area` is called below? But I'm passing `datasets` to that call too.

Reply:
Oh indeed you are, sorry!

Reply:
So the transformation do `id` isn't necessary anymore?

Reply:
Correct. In both cases (`finest_area` and `copy`) the list of `datasets` is passed to `Scene.__getitem__` so it should handle all of the complexity of what DataID matches with the provided name/wavelength/DataQuery/DataID.